minigui/README.md

```diff
@@ -53,36 +69,247 @@ A UI for Minigo
+result. Additionally, when processing a `loadsgf` message, an `mg-position`
+object should be printed after playing each move in the SGF.
+
+Below is a example from Minigo. This example has been formatted nicely to make it
+more legible, but the engine **must** print the JSON object to a single line
+otherwise Minigui will not be able to parse it.
+
+```
+mg-position: {
+  "gameOver": false,
+  "id": "0x559687366c10",
+  "move": "G10",
+  "moveNum": 36,
+  "parentId": "0x559687364820",
+  "stones": "........................................O.O...............XX....X.......X....OXO................XOO................XXO.................XO.................XXOO................OXXO.......................................................................X...............................O......O........X.X.O.......O..X....XOO.........................................",
+  "toPlay": "B"
+}
+```
+
+And the stones in this board position are:
+
+```
+. . . . . . . . . . . . . . . . . . .
+. . . . . . . . . . . . . . . . . . .
+. . O . O . . . . . . . . . . . . . .
+. X X . . . . X . . . . . . . X . . .
+. O X O . . . . . . . . . . . . . . .
+. X O O . . . . . . . . . . . . . . .
+. X X O . . . . . . . . . . . . . . .
+. . X O . . . . . . . . . . . . . . .
+. . X X O O . . . . . . . . . . . . .
+. . . O X X O . . . . . . . . . . . .
+. . . . . . . . . . . . . . . . . . .
+. . . . . . . . . . . . . . . . . . .
+. . . . . . . . . . . . . . . . . . .
+. . X . . . . . . . . . . . . . . . .
+. . . . . . . . . . . . . . . O . . .
+. . . O . . . . . . . . X . X . O . .
+. . . . . O . . X . . . . X O O . . .
+. . . . . . . . . . . . . . . . . . .
+. . . . . . . . . . . . . . . . . . .
+```
+
+And here's how an engine should print the position JSON: all one one line with a
+`mg-position:` prefix.
+
+```
+mg-position: {"gameOver":false,"id":"0x559687366c10","move":"G10","moveNum":36,"parentId":"0x559687364820","stones":"........................................O.O...............XX....X.......X....OXO................XOO................XXO.................XO.................XXOO................OXXO.......................................................................X...............................O......O........X.X.O.......O..X....XOO.........................................","toPlay":"B"}
+```
+
+#### Update messages
+
+While an engine is thinking (in response to a `genmove` command or pondering
+while it's the other player's turn), an engine should also send incremental
+updates for the current position, containing variations considered, expected win
+rate, number of visits, etc.
+
+These updates should be printed as a JSON object (again on a single line) with
+the prefix `mg-update:`.
+
+ - `id: string`: the position's id.
+ - `n?: number`: the number of reads for the position.
+ - `q?: number`: estimated winrate for the position from black's perspective in
+   the range [-1, 1].
+ - `pv?: string`: the GTP coordinate of the variation in `variations` that
```

Comment:
Nit: consider adding something like "E.g., the coordinate "K10" for the variation beginning at K10..."

Reply:
esp as the PV is missing from below.

Reply:
done. actually the updates don't even have a `pv` field any more but I forgot to update the interface definition the typescript code